lom/src/lib/lomtelemetry/client.go

```diff
@@ -0,0 +1,368 @@
+package lomtelemetry
+
+import (
+    "encoding/json"
+    cmn "lom/src/lib/lomcommon"
+    "time"
+)
+
+/*
+ * GetPubChannel
+ *
+ * Get channel for publishing events, counters, red-button, ...
+ * Once opened it stays till system shutdown or terminates upon
+ * i/p data channel close.
+ * NOTE: The Pub channel can be opened only once per process. Any
+ *       pre-mature termination will block any further publish.
+ *
+ * Closing returned channel chData will close the underlying
+ * network connection.
+ *
+ * Input:
+ *  chtype      - Type of data to be published
+ *  producer    - Is this engine, pluginMgr or plugin
+ *  suffix      - Producer suffix. Say plugin-name in case of plugin.
+ *
+ * Output: None
+ *
+ * Return:
+ *  chData      - Input data channel for publishing. All data written
+ *                into this channel by anyone are published.
+ *                Expect a JSON string.
+ *                Closing this shuts down this pubchannel
+ *  err - Any error
+ *
+ */
+const PUB_BUFFER_CNT = 100 /* Allow publishers for short bursts */
+
+func GetPubChannel(chtype ChannelType_t, producer ChannelProducer_t,
+    pluginName string) (chData chan<- JsonString_t, err error) {
+
+    ch := make(chan JsonString_t, PUB_BUFFER_CNT)
+
+    defer func() {
+        if err != nil {
+            close(ch)
+            ch = nil
+        }
+    }()
+
+    prefix := ""
+    if prefix, err = GetProdStr(producer, pluginName); err != nil {
+        /* err is detailed enough */
+    } else if err = openPubChannel(chtype, prefix, ch); err != nil {
+        err = cmn.LogError("Failed to get pub channel (%v)", err)
+    } else {
+        chData = ch
+    }
+    return
+}
+
+/*
+ * GetSubChannel
+ *
+ * Get channel for subscribing from other processes.
+ * Runs until system shutdown.
+ *
+ * Input:
+ *  chtype - Type of data to receive.
+ *  producer - If interested from a specific producer, specify, else receive from all.
+ *  pluginName - If producer is a plugin, its name
```

Comment:
_"If producer is a plugin, its name"_
GetSubChannel - this method will be used only to subscribe for red button status right ?

Reply:
It could be used by a CLI command/tool too.

I want to see only data published by this plugin
Though Engine is the main publisher for anomaly events, it does not block plugin to make some private publish.